src/main/java/com/extendedclip/papi/expansion/player/PlayerExpansion.java

```diff
@@ -21,469 +21,336 @@
 import me.clip.placeholderapi.expansion.PlaceholderExpansion;
-import org.bukkit.Bukkit;
-import org.bukkit.ChatColor;
-import org.bukkit.Material;
-import org.bukkit.OfflinePlayer;
-import org.bukkit.World;
-import org.bukkit.enchantments.Enchantment;
+import me.clip.placeholderapi.expansion.Taskable;
+import org.bukkit.*;
+import org.bukkit.block.BlockFace;
+import org.bukkit.configuration.ConfigurationSection;
 import org.bukkit.entity.Player;
+import org.bukkit.event.HandlerList;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.bukkit.potion.PotionEffect;
 import org.bukkit.potion.PotionEffectType;
 
+import java.text.SimpleDateFormat;
 import java.util.*;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
-import static com.extendedclip.papi.expansion.player.PlayerUtil.durability;
-import static com.extendedclip.papi.expansion.player.PlayerUtil.format12;
-import static com.extendedclip.papi.expansion.player.PlayerUtil.format24;
-import static com.extendedclip.papi.expansion.player.PlayerUtil.getBiome;
-import static com.extendedclip.papi.expansion.player.PlayerUtil.getCapitalizedBiome;
-import static com.extendedclip.papi.expansion.player.PlayerUtil.getEmptySlots;
-import static com.extendedclip.papi.expansion.player.PlayerUtil.getDirection;
-import static com.extendedclip.papi.expansion.player.PlayerUtil.getTotalExperience;
-import static com.extendedclip.papi.expansion.player.PlayerUtil.getXZDirection;
-import static com.extendedclip.papi.expansion.player.PlayerUtil.itemInHand;
-
-public final class PlayerExpansion extends PlaceholderExpansion implements Configurable {
-    private String low;
-    private String medium;
-    private String high;
-
-    private int mediumValue;
-    private int highValue;
-
-    private String north;
-    private String northEast;
-    private String east;
-    private String southEast;
-    private String south;
-    private String southWest;
-    private String west;
-    private String northWest;
 
-    @Override
-    public String getIdentifier() {
-        return "player";
+public final class PlayerExpansion extends PlaceholderExpansion implements Taskable, Configurable {
+
+    @Getter private final String identifier = "player";
+    @Getter private final String author = "clip";
+    @Getter private final String version = "2.1.0";
+    @Getter private final Map<String, Object> defaults;
+
+    private final SimpleDateFormat dateFormat = PlaceholderAPIPlugin.getDateFormat();
+    private final VersionHelper versionHelper;
+    private final PlayerListener listener;
+
+    private final Map<String,String> pingColors = new HashMap<>();
+    private final Map<BlockFace,String> directions = new HashMap<>();
+    final Map<Player, Long> joinTimes = new HashMap<>();
+    final Map<Player,Double> damagesGiven = new HashMap<>();
+
+    public PlayerExpansion() {
+        defaults = new HashMap<>() {{
+            put("ping",new HashMap<>() {{
```

Comment:
There is ImmutableMap#Builder from guava, please avoid using anonymous classes like this.